### ISSUE TITLE
[REQUEST] Diagnostic regression tests

### ISSUE THREAD

**Issue by lilyminium** [openforcefield/openff-toolkit#1051]
**Is your feature request related to a problem? Please describe.**

Some time ago, the charge model changed from ELF10 to single-conformer AM1BCC without many people noticing, possibly unintentionally. This seems dangerous.

**Describe the solution you'd like**

A regression test that tests for specific "default" behaviour. I know this does not currently exist because I changed the charge model back to ELF10 and all tests passed. That means, if the model should change again, even those checking over the PR may not be fully aware. https://github.com/openforcefield/openff-toolkit/commit/60ce9594a7f79b766d1359809c9f52163abdbcae

**Describe alternatives you've considered**

Perhaps the absence of failing tests means that the actual charge differences are negligible, at least in the test cases currently in the test suite. However, it would still be good for any changes to the default settings to require *some* change to the tests.

**Additional context**

**Comment by mattwthompson** [comment]
> actual charge differences [might be] negligible

From your tinkering you have a sense of what these differences numerically tend to be?

**Comment by mattwthompson** [comment]
FTR we added some basic energy tests in https://github.com/openforcefield/openff-toolkit/commit/14c963fae0bc2e12d328070eb726a7848f87cf25  as part of the saga resolved in #808. This happened before #722; it's possible but not guaranteed that these would have caught that change. I don't know if differences in partial charges would be enough to be picked up there.

The scope of `test_energies.py` is small, but should in principle catch a lot of behavior changes. Still, I'd strongly be in favor of expanding it to cover things it currently misses.

**Comment by davidlmobley** [comment]
I strongly agree that we want diagnostic regression tests, such as those that would have caught this. Certainly sudden changes in charges of 0.01 or 0.1 e ought to result in issues, or energy changes of a couple of tenths of a kcal/mol ought to be caught. 

Some code changes will perhaps result in larger deviations than this that are OK, but we would want to know about them and assess, meaning that we really ought to have somehting in place which is flagging them for us.

**Comment by mattwthompson** [comment]
I'm working on a set of regression tests in [another repo](https://github.com/mattwthompson/interchange-regression-testing), which is a part of the Interchange adoption [plan](https://openforcefield.atlassian.net/wiki/spaces/IN/pages/2225111061/Project+Plan%3A+Adoption+of+Interchange+backend+as+export+machinery+for+ForceField.create_openmm_system?atlOrigin=eyJpIjoiMTJhY2RkNjA0ODBhNGI4ZWI2ZmM3ZjMzNzgzMDZhNjEiLCJwIjoiY29uZmx1ZW5jZS1jaGF0cy1pbnQifQ). Despite the naming, it's toolkit-centric and meant to test results from different conda environments, presumably with different versions of the toolkit installed. It's a work in progress and will change a few times before we want to have much faith in it, but once online it could be molded into a pre-release turnkey. A notable difference is that is mostly relies on comparing serialized representations of the OpenMM `System`s; this should generally be a higher standard to meet than energy differences but there may be subtle differences between the approaches.

A minor annoyance is that because it's meant to test different versions of the toolkit, it compares against different Python calls, it absorbs the problems of run-to-run variance of wrapped methods.  This is mostly AM1-BCC charges but can sometimes affect other things like fractional bond orders.

**Comment by mattwthompson** [comment]
It might be useful to put some bounds on what behavior we really must avoid regressions with, or at least split things up. If it's a matter of the charges/conformers/etc. that result from various `Molecule` API calls, that'll be hard for us to take responsibility for given the number of ways things may differ run-to-run or with small changes upstream. Results from OpenMM system creation can get dubious depending on the standards that are used ... we put a good bit of work into https://github.com/openforcefield/interchange-regression-testing and I'm currently trying to add automation to that repo and a subset of those tests in the Interchange test suite. "Default" behavior is hard to get a handle on given the number of things that users might do, and there's also the question of "default behavior of what?"

From the other perspective, we could build things from the ground up one by one. That list of tests would be curated first from issues we've had in the past and then expanded to things we're generally worried about.
